sample-apps/README.md

```diff
@@ -0,0 +1,41 @@
+# What is a Grafana Cloud Integration, and what are sample-apps supporting them?
+A Grafana Cloud Integration is a prebuilt package of dashboards, recording rules, alerts, metadata, and scrape configurations intended to make it quick to add monitoring to a number of systems. See here for more information on [Grafana Cloud Integrations](https://grafana.com/docs/grafana-cloud/what-are/integrations/) and the [Integrations reference](https://grafana.com/docs/grafana-cloud/monitor-infrastructure/integrations/integration-reference/) for a list of supported systems.
+
+Sample-apps are pre-configured applications and load-testing that allows us to spin up a system for a given integration and quickly validate the metrics emitted match what we expect for our Cloud Integrations.
+
+# Adding CI support to a Multipass based sample-app
+Please note, this is not a full tutorial in creating sample-apps from scratch.
+
+Sample-apps are run in CI using Multipass on `ubuntu-latest-8-core`, with both Loki and Mimir spun up outside the sample-app. Nested virtualisation is available, and multiple VMs can be created as desired.
+
+**Workflow timeout is currently set at 30 minutes, and will fail if exceeded**
+
+The CI workflow expects a few things from a CI enabled sample app. For a working example, please check out the [openldap sample-app](/sample-apps/openldap/):
+
+1. Ensure the sample-app is built using Multipass
+   - Sample should be stand-alone and not contain any databases
+1. Ensure sample-app has load-generation scripts or capabilities to accurately produce metrics for the majority of use-cases
+1. Ensure sample-app name matches integration slug
+1. Add an `expected_metrics` file to the sample-app dir, containing a newline separated list of Prometheus metrics expected by the integration, e.g:
+    ```
+    up
+    metric_name_1
+    metric_name_2
```

Comment:
Could you metrics with expected labels? i.e. metric_nam_1{label1="value"}

Reply:
Not at this time no, though that can easily be added as a separate test suite in a second step!
The intent here is not so much to check labels, or anything specifically with regards to them, but more to check if all the metrics expected by the dashboards in the integration are present.

The `expected_metrics` file is equivalent to the `metric_names` files found within our cloud integrations in their repository, and an upcoming PR will automatically update them if matching sample apps are found.

As such, it's only required to be added manually when first adding a new sample app.

Also, in this case, I've included `metric_name_1` and `metric_name_2` manually to test that it still passes with a >= 80% match rate. They will be removed in a subsequent PR